airflow/io/path.py

```diff
@@ -198,7 +198,7 @@ def replace(self, target) -> ObjectStoragePath:
 
         Returns the new Path instance pointing to the target path.
         """
-        return self.rename(target, overwrite=True)
+        return self.rename(target)
```

Comment:
@bolkedebruin @ap-- -> mypy complains and there does not seem to be overwrite kwarg when you look at the code . I am not sure if there was a reason it was added in a first place - maybe it is needed somehow to pass down to unrelying implementations? 

Please advice if this is a good idea to remove it.

Reply:
If @bolkedebruin is not able to respond then we could merge and leave a link with the bug ticket behind?